src/test/java/org/sagebionetworks/bridge/spring/controllers/CRCControllerTest.java

```diff
@@ -0,0 +1,671 @@
+import org.sagebionetworks.bridge.services.ParticipantService;
+import org.sagebionetworks.bridge.services.ReportService;
+import org.sagebionetworks.bridge.services.SessionUpdateService;
+
+public class CRCControllerTest extends Mockito {
+
+    static final LocalDate JAN1 = LocalDate.parse("1970-01-01");
+    static final LocalDate JAN2 = LocalDate.parse("1970-01-02");
+    static final String HEALTH_CODE = "healthCode";
+    String CREDENTIALS = USERNAME + ":dummy-password";
+    String AUTHORIZATION_HEADER_VALUE = "Basic "
+            + new String(Base64.getEncoder().encode(CREDENTIALS.getBytes()));
+    
+    static final AccountSubstudy ACCT_SUB1 = AccountSubstudy.create(APP_ID, "substudyA", USER_ID);
+    static final AccountSubstudy ACCT_SUB2 = AccountSubstudy.create(APP_ID, "substudyB", USER_ID);
+    static final AccountId ACCOUNT_ID = AccountId.forId(APP_ID, USER_ID);
+    
+    @Mock
+    ParticipantService mockParticipantService;
+    
+    @Mock
+    ReportService mockReportService;
+    
+    @Mock
+    AppService mockAppService;
+    
+    @Mock
+    AccountService mockAccountService;
+    
+    @Mock
+    SessionUpdateService mockSessionUpdateService;
+    
+    @Mock
+    App mockApp;
+    
+    @Mock
+    HttpServletRequest mockRequest;
+    
+    @Mock
+    HttpServletResponse mockResponse;
+    
+    @Captor
+    ArgumentCaptor<SignIn> signInCaptor;
+    
+    @Captor
+    ArgumentCaptor<StudyParticipant> participantCaptor;
+    
+    @Captor
+    ArgumentCaptor<ReportData> reportCaptor;
+    
+    @Captor
+    ArgumentCaptor<Account> accountCaptor;
+    
+    App app;
+    
+    Account account;
+    
+    @InjectMocks
+    @Spy
+    CRCController controller;
+
+    @BeforeMethod
+    public void beforeMethod() {
+        MockitoAnnotations.initMocks(this);
+
+        account = Account.create();
+        account.setHealthCode(HEALTH_CODE);
+        account.setAccountSubstudies(ImmutableSet.of(ACCT_SUB1, ACCT_SUB2));
+        account.setDataGroups(ImmutableSet.of("group1"));
+        
+        app = App.create();
+        when(mockAppService.getApp(APP_ID)).thenReturn(app);
+        
+        doReturn(mockRequest).when(controller).request();
+        doReturn(mockResponse).when(controller).response();
+        doReturn(TestConstants.TIMESTAMP).when(controller).getTimestamp();
+    }
+    
+    @AfterMethod
+    public void afterMethod() {
+        BridgeUtils.setRequestContext(RequestContext.NULL_INSTANCE);
+    }
+    
+    @Test
+    public void updateParticipantCallsExternal() throws Exception {
+        StudyParticipant participant = new StudyParticipant.Builder()
+                .withDataGroups(makeSetOf(CRCController.AccountStates.SELECTED, "group1")).build();
+        mockRequestBody(mockRequest, participant);
+        
+        UserSession session = new UserSession();
+        session.setAppId(APP_ID);
+        session.setParticipant(new StudyParticipant.Builder().withId(USER_ID).build());
```

Comment:
Should use a different USER_ID for the researcher for clarity. (For both versions of the test.)

Reply:
This is definitely a bug. Will also add a test of this method in the integration tests (and fixed it here).

Reply:
Now uses a different user id.